src/crewai/agents/agent_adapters/foundry_agents/foundry_adapter.py

```diff
@@ -0,0 +1,184 @@
+from typing import Any, List, Optional
+import os
+import time
+import json
+from pydantic import Field, PrivateAttr
+
+from crewai.agents.agent_adapters.base_agent_adapter import BaseAgentAdapter
+from crewai.agents.agent_adapters.foundry_agents.structured_output_converter import (
+    FoundryConverterAdapter,
+)
+from crewai.agents.agent_builder.base_agent import BaseAgent
+from crewai.tools import BaseTool
+from crewai.tools.agent_tools.agent_tools import AgentTools
+from crewai.utilities import Logger
+from crewai.utilities.events import crewai_event_bus
+from crewai.utilities.events.agent_events import (
+    AgentExecutionCompletedEvent,
+    AgentExecutionErrorEvent,
+    AgentExecutionStartedEvent,
+)
+
+try:
+    from azure.ai.projects import AIProjectClient as FoundryClient
+    from azure.ai.projects.models import MessageTextContent
+    from azure.identity import DefaultAzureCredential
+    from azure.ai.projects.models import FunctionTool
+
+
+    from .foundry_agent_tool_adapter import FoundryAgentToolAdapter
+
+    FOUNDRY_AVAILABLE = True
+except ImportError:
+    FOUNDRY_AVAILABLE = False
+
+
+class FoundryAgentAdapter(BaseAgentAdapter):
+    """Adapter for Foundry Assistants"""
+
+    model_config = {"arbitrary_types_allowed": True}
+
+    _foundry_client: "FoundryClient" = PrivateAttr()
+    _logger: Logger = PrivateAttr(default_factory=lambda: Logger())
+    _active_thread_id: Optional[str] = PrivateAttr(default=None)
+    function_calling_llm: Any = Field(default=None)
+    step_callback: Any = Field(default=None)
+    _tool_adapter: "FoundryAgentToolAdapter" = PrivateAttr()
+    _converter_adapter: FoundryConverterAdapter = PrivateAttr()
+    _converted_tools: Optional[FunctionTool] = PrivateAttr(default=None)
+
+    def __init__(
+        self,
+        model: str = "gpt-4.1-mini",
+        tools: Optional[List[BaseTool]] = None,
+        agent_config: Optional[dict] = None,
+        **kwargs,
+    ):
+        if not FOUNDRY_AVAILABLE:
+            raise ImportError(
+                "Foundry Agent Dependencies are not installed. Please install it using `uv pip install azure-ai-projects azure-identity`"
+            )
+        else:
```

Comment:
You can drop the `else` since you are throwing an error if `not FOUNDRY_AVAILABLE`